k8s.go

```diff
@@ -252,3 +307,98 @@ func secretToString(rawSecret []byte, doubleBase64 bool) (string, error) {
 
 	return content, nil
 }
+
+func getConfigMapK8s(client *kubernetes.Clientset,
+	namespace, name string) (*api.ConfigMap, bool, error) {
+
+	log("Attempting to load configmap %s from namespace %s", name, namespace)
+	configMap, err := client.CoreV1().ConfigMaps(namespace).Get(
+		context.Background(), name, metav1.GetOptions{},
+	)
+
+	switch {
+	case err == nil:
+		log("ConfigMap %s loaded successfully", name)
+		return configMap, true, nil
+	case errors.IsNotFound(err):
+		log("ConfigMap %s not found in namespace %s", name, namespace)
+		return nil, false, nil
+	default:
+		return nil, false, fmt.Errorf("error querying configmap existence: %v", err)
```

Comment:
nit: some lines in this file are still too long. Should attempt to break everything before 80 characters (with the exception of the struct tags with back ticks, since those can't easily be wrapped).

Reply:
🔭 Thanks for the guidance here. Re-configured the vertical ruler in IDE so it's easier to spot this.